File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int numberOfDigits = 1 + (int)Math.log10(numberToCheck);

        int n = numberToCheck;
        int sum = 0;

        while (n > 0) {
            int digit = n % 10;
            sum += Math.pow(digit, numberOfDigits);
            n /= 10;
        }

        return sum == numberToCheck;
    }

}

File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate)
{
    int number = candidate;
    int sum = 0;
    int power = (int) log10(number) + 1;

    while (number > 0) {
        int digit = number % 10;
        sum += pow(digit, power);
        number /= 10;
    }

    return sum == candidate;
}